packages/state-manager/src/sagas/publicChannels/sendUnregisteredInfoMessage/sendUnregisteredInfoMessage.saga.ts

```diff
@@ -0,0 +1,25 @@
+import { put, select, call } from 'typed-redux-saga'
+import { messagesActions } from '../../messages/messages.slice'
+import { publicChannelsSelectors } from '../publicChannels.selectors'
+import { WriteMessagePayload, MessageType } from '@quiet/types'
+import { communitiesSelectors } from '../../communities/communities.selectors'
+import { identitySelectors } from '../../identity/identity.selectors'
+import { userJoinedMessage } from '@quiet/common'
+
+export function* sendUnregisteredInfoMessage(): Generator {
+  const community = yield* select(communitiesSelectors.currentCommunity)
+  const identity = yield* select(identitySelectors.currentIdentity)
+  const generalChannel = yield* select(publicChannelsSelectors.generalChannel)
+
+  if (!community?.name || !identity || !generalChannel) return
```

Comment:
Don't we risk not sending this 'welcome' message at all, depending on the race condition of replicating channels data and entering saga? I guess what we need is to wait for actual general channel replciation before entering the saga rather than exiting in case of lack of general channel data in store